react-components/src/hooks/useSearchAssetsMapped360Annotations.tsx

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+
+import {
+  type AnnotationFilterProps,
+  type FileFilterProps,
+  type AnnotationsBoundingVolume,
+  type Asset,
+  type CogniteClient,
+  type IdEither,
+  type AnnotationModel
+} from '@cognite/sdk';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+import { chunk, uniqBy } from 'lodash';
+
+export const useAllAssetsMapped360Annotations = (
+  siteIds: string[],
+  limit: number = 100,
+  sdk: CogniteClient
+): UseQueryResult<Asset[]> => {
+  return useQuery(
+    ['reveal', 'react-components', 'all-assets-mapped-360-annotations', siteIds],
+    async () => {
+      const assetMappings = await getAssetsMapped360Annotations(sdk, siteIds);
+      return assetMappings;
+    },
+    {
+      staleTime: Infinity
+    }
+  );
+};
+
+export const useSearchAssetsMapped360Annotations = (
+  siteIds: string[],
+  sdk: CogniteClient,
+  query: string
+): UseQueryResult<Asset[]> => {
+  return useQuery(
+    ['reveal', 'react-components', 'search-assets-mapped-360-annotations', query, siteIds],
+    async () => {
+      if (query === '') {
+        const assetMappings = await getAssetsMapped360Annotations(sdk, siteIds);
+        return assetMappings;
+      }
+      const assetMappings = await getAssetsMapped360Annotations(sdk, siteIds);
+
+      const filteredSearchedAssets =
+        assetMappings.filter((asset) => {
+          const isInName = asset.name.toLowerCase().includes(query.toLowerCase());
+          const isInDescription = asset.description?.toLowerCase().includes(query.toLowerCase());
+
+          return isInName || isInDescription;
+        }) ?? [];
+
+      return filteredSearchedAssets;
+    },
+    {
+      staleTime: Infinity
+    }
+  );
+};
+
+async function getAssetsMapped360Annotations(
+  sdk: CogniteClient,
+  siteIds: string[]
+): Promise<Asset[]> {
+  const fileIdsList = await get360ImagesFileIds(siteIds, sdk);
+  const image360Annotations = await get360ImageAnnotations(fileIdsList, sdk);
+  const result = await get360AnnotationAssets(image360Annotations, sdk);
+
+  return result;
+}
+
+async function get360AnnotationAssets(
+  image360Annotations: AnnotationModel[],
+  sdk: CogniteClient
+): Promise<Asset[]> {
+  const annotationMapping = image360Annotations.map((annotation) => ({
+    assetId:
+      ((annotation.data as AnnotationsBoundingVolume).assetRef?.id as number) ??
```

Comment:
What happens when assetRef is undefined? We should handle this case as well. I think type conversion to number and string shouldn't be used here.